src/kinetics/ChebyshevRate.cpp

```diff
@@ -112,14 +109,20 @@ void ChebyshevRate::setLimits(double Tmin, double Tmax, double Pmin, double Pmax
 
 void ChebyshevRate::setData(const Array2D& coeffs)
 {
-    m_coeffs = coeffs;
-    dotProd_.resize(coeffs.nRows());
+    m_valid = coeffs.data().size();
```

Comment:
Does this type coercion result in warnings from the compiler on any major platforms? Elsewhere, this similar condition was expressed with `.empty()`, which may be better here.

Reply:
Casting from an integer to a boolean is understood by all compilers, but I don't have a problem using `.empty()`.